### PR TITLE
Add support for spacy >= 2.2 

### PR DESCRIPTION
**Pull request by tamuhey**
from #3424

- changed requirements: `spacy < 2.2` => `spacy < 2.3`
- modified some tests. They fail due to chages of spacy's statistical models

### CODE REVIEW

File: allennlp/tools/EVALB/evalb.dSYM/Contents/Info.plist [hunk 1]
```diff
@@ -1,5 +1,5 @@
 <?xml version="1.0" encoding="UTF-8"?>
-<!DOCTYPE plist PUBLIC "-//Apple Computer//DTD PLIST 1.0//EN" "https://www.apple.com/DTDs/PropertyList-1.0.dtd">
```
Comment (DeNeutoy) [inline allennlp/tools/EVALB/evalb.dSYM/Contents/Info.plist]: This diff looks unrelated, could you remove it?

**Comment by tamuhey** [comment]
@DeNeutoy I reverted the file. What else should I do?

**Comment by DeNeutoy** [comment]
nothing! This looks good to go 👍, thanks for the effort.

**Comment by matt-gardner** [comment]
This is now causing master to fail tests.  I vote we try to move away from tests that explicitly rely on the specifics of spacy output, as it's not stable enough for us.
